homeassistant/components/stream/__init__.py

```diff
@@ -572,6 +573,19 @@ def get_diagnostics(self) -> dict[str, Any]:
         """Return diagnostics information for the stream."""
         return self._diagnostics.as_dict()
 
+    @staticmethod
+    async def async_has_stream_auth_error(hass: HomeAssistant, source: str) -> bool:
```

Comment:
I don't think this needs to be on `Stream` since its not really related to it. How about it's a top level function?

Reply:
Done